velox/functions/sparksql/tests/SplitTest.cpp

```diff
@@ -0,0 +1,282 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "velox/common/base/tests/GTestUtils.h"
+#include "velox/functions/sparksql/tests/SparkFunctionBaseTest.h"
+#include "velox/type/Type.h"
+
+using namespace facebook::velox::test;
+namespace facebook::velox::functions::sparksql::test {
+namespace {
+
+class SplitTest : public SparkFunctionBaseTest {
+ protected:
+  void testSplit(
+      const std::vector<std::string>& input,
+      std::string delim,
+      std::optional<int32_t> limit,
+      size_t numRows,
+      const std::vector<std::vector<std::string>>& expected) {
+    auto strings = makeFlatVector(input);
+    auto delims = makeConstant<StringView>(StringView{delim}, numRows);
+    auto expectedResult = makeArrayVector(expected);
+    VectorPtr result;
+    if (limit.has_value()) {
+      auto limits = makeConstant<int32_t>(limit.value(), numRows);
+      auto input = makeRowVector({strings, delims, limits});
+      result = evaluate("split(c0, c1, c2)", input);
+    } else {
+      auto input = makeRowVector({strings, delims});
+      result = evaluate("split(c0, c1)", input);
+    }
+    assertEqualVectors(expectedResult, result);
+  };
+};
+
+TEST_F(SplitTest, basic) {
+  auto limit = -1;
```

Comment:
Wondering if the test will be more readable if this variable is removed and its usage is replaced with hard-coded values.

```
  // Negative limit.
  testSplit(input, delim, -1, numRows, expected);

  // Limit greater than the input string size.
  testSplit(input, delim, 100, numRows, expected);
```

Reply:
got! Remove all `limit` and `delim` variable and replace with hardcode values.